packages/xo-web/src/xo-app/new-vm/index.js

```diff
@@ -715,7 +719,10 @@ export default class NewVm extends BaseComponent {
       resolveIds(
         filter(
           this._getResolvedResourceSet().objectsByType.SR,
-          this._getSrPredicate()
+          sr =>
+            sr.$pool === template.$pool &&
+            sr.content_type !== 'iso' &&
+            sr.size > 0
```

Comment:
The SR must still be in the resource set though, right?

Reply:
it's an `SR` from resource set

Reply:
Why don't you keep `this._getSrPredicate`, here?